strawberryfields/gbs/plot.py

```diff
@@ -252,11 +259,94 @@ def plot_subgraph(subgraph: nx.Graph, size: int = 500) -> None:  # pragma: no co
         xaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
         yaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
         margin=dict(b=0, l=0, r=0, t=25),
-        height=size,
-        width=size,
+        height=plot_size,
+        width=plot_size,
         plot_bgcolor="#ffffff",
     )
 
     f = go.Figure(data=[g_edges, g_nodes], layout=layout)
 
     return f
+
+
+def plot_points(R: np.ndarray, sample: Optional[list] = None,
+                plot_size: int = 500, point_size: int = 30) -> None:  # pragma: no cover
+    """Creates a Plotly plot of two-dimensional points given their input coordinates. Sampled
+    points can be optionally highlighted among all points.
+
+    **Example usage:**
+
+    >>> R = np.random.normal(0, 1, (50, 2))
+    >>> sample = [1] * 10 + [0] * 40  # select first ten points
+    >>> plot_points(R, sample).show()
+
+    .. image:: ../../_static/normal_pp.png
+       :width: 40%
+       :align: center
+       :target: javascript:void(0);
+
+    Args:
+        R (np.array): Coordinate matrix. Rows of this array are the coordinates of the points.
+        sample (list[int]): optional subset of sampled points to be highlighted
+
+    Returns:
+         Figure: Plotly figure of points with the sample highlighted
+    """
+    try:
+        import plotly.graph_objects as go
+    except ImportError:
+        raise ImportError(
+            "Plotly required for using plot_points(). Can be installed using pip install "
+            "plotly or visiting https://plot.ly/python/getting-started/#installation"
+        )
+    except RuntimeError:
+        print("Plotly unable to open display")
+        raise
```

Comment:
This `except:` is a bit weird. If plotly already raises an error when not able to open a display, why do we need to catch it?

Reply:
Also, since this `try/except` statement occurs in multiple places, might as well collate it to a single function call (such as `try_import_plotly()`).

Reply:
Yeah I'd be tempted to go with just the `ImportError` since I'm not so confident about the `RuntimeError` and if that is the thing that's triggered when Plotly can't open a display

Reply:
Good catch @josh146, I'm happy with keeping only the `ImportError` and grouping it into a single function